state/index.js

```diff
@@ -0,0 +1,92 @@
+var generators = require('yeoman-generator');
+var chalk = require('chalk');
+var fs = require('fs');
+var program = require('ast-query');
+var readFileAsString = require("html-wiring").readFileAsString;
+
+var prompts = [{
+  type: 'confirm',
+  name: 'createAsSubfolder',
+  message: 'Create state slice in its own folder in state/?',
+  default: true
+},
+  {
+    type: 'confirm',
+    name: 'updateIndex',
+    message: 'Add new slice to state/index.js (if it already exists)?',
+    default: true
+  }];
+
+function normalizeName(name) {
+  return name.split(/(?=[A-Z])/)
```

Comment:
What do you think about abstracting these 4 helper methods out into a utils.js file in the root dir and just requiring it here?  They could potentially be helpful in other places.  I'm also seeing other things that would be better in a single place (such as the Visual Studio warning).  It might be nice to have a central place for all of that to live.